.github/scripts/generate_ci_workflows.py

```diff
@@ -335,7 +335,7 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
     ),
     CIWorkflow(
         arch="linux",
-        build_environment="linux-xenial-cuda11.1-py3.6-gcc7",
+        build_environment="trunk-only-linux-xenial-cuda11.1-py3.6-gcc7",
```

Comment:
is this one also trunk only? if so, need to add `trigger_action_only=True` to the CIFlowConfig

Reply:
also need to remove `ciflow/default` to make it not run by default on PR

![image](https://user-images.githubusercontent.com/658840/129960488-4df51ef7-5429-47c3-a538-56b38f5190f7.png)

Reply:
this might've been changed recently (i dont think we previously have a cuda linux run on PR) but it might've been a purposeful change.

Reply:
i will stop merging this until we have a clear picture regarding how to auto-gen these trunk-only and periodic names. as @seemethere mentioned, it is tedious to maintain them manually. 

My suggestion is to leverage ciflow configurations. I will sync with @zhouzhuojie offline to see what's the best auto-marker for them